File: code/10-state-space-search/python/graph_coloring_demo.py
from state_space_search import color_graph

if __name__ == "__main__":
    # A-B-C-D-A cycle plus A-C diagonal, matching the lecture's own example.
    adjacency = [
        [0, 1, 1, 1],  # A: B, C, D
        [1, 0, 1, 0],  # B: A, C
        [1, 1, 0, 1],  # C: A, B, D
        [1, 0, 1, 0],  # D: A, C
    ]
    colors = color_graph(adjacency, 3)
    print("colors:", ",".join(str(c) for c in colors))

File: code/10-state-space-search/c/graph_coloring_demo.c
#include "state_space_search.h"
#include <stdio.h>

int main(void) {
    /* A-B-C-D-A cycle plus A-C diagonal, matching the lecture's own example. */
    const unsigned char adjacency[16] = {
        0,1,1,1,
        1,0,1,0,
        1,1,0,1,
        1,0,1,0,
    };
    int colors[4];
    if (ss_color_graph(adjacency, 4, 3, colors) != SS_OK) return 1;
    printf("colors: %d,%d,%d,%d\n", colors[0], colors[1], colors[2], colors[3]);
    return 0;
}
